crypto/ocsp/ocsp_lib.c

```diff
@@ -0,0 +1,117 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ocsp_internal.h"
+
+OCSP_CERTID *OCSP_cert_to_id(const EVP_MD *dgst, const X509 *subject,
+                             const X509 *issuer)
+{
+  if (issuer == NULL){
+    OPENSSL_PUT_ERROR(OCSP, ERR_R_PASSED_NULL_PARAMETER);
+    return NULL;
+  }
+
+  const X509_NAME *iname;
+  const ASN1_INTEGER *serial;
+  ASN1_BIT_STRING *ikey;
+
+  if (dgst == NULL) {
+    dgst = EVP_sha1();
+  }
+  if (subject != NULL) {
+    iname = X509_get_issuer_name(subject);
+    serial = X509_get0_serialNumber(subject);
+  } else {
+    iname = X509_get_subject_name(issuer);
+    serial = NULL;
+  }
+  ikey = X509_get0_pubkey_bitstr(issuer);
+  return OCSP_cert_id_new(dgst, iname, ikey, serial);
+}
+
+OCSP_CERTID *OCSP_cert_id_new(const EVP_MD *dgst,
+                              const X509_NAME *issuerName,
+                              const ASN1_BIT_STRING *issuerKey,
+                              const ASN1_INTEGER *serialNumber)
+{
+  if(dgst == NULL || issuerName == NULL || issuerKey == NULL){
+    OPENSSL_PUT_ERROR(OCSP, ERR_R_PASSED_NULL_PARAMETER);
+    return NULL;
+  }
+
+  int nid;
+  unsigned int i;
+  X509_ALGOR *alg;
+  OCSP_CERTID *cid = NULL;
+  unsigned char md[EVP_MAX_MD_SIZE];
+
+  if ((cid = OCSP_CERTID_new()) == NULL) {
+    goto err;
+  }
+
+  alg = cid->hashAlgorithm;
+  ASN1_OBJECT_free(alg->algorithm);
```

Comment:
`alg` can be `NULL`?

Reply:
The alg structure is created by the OCSP_CERTID_new ASN1 function above, so I assume it will be initialized for use already.